asab/web/tenant/service.py

```diff
@@ -1,45 +1,75 @@
-import asab
+import asyncio
+import aiohttp
+
+from ...abc.service import Service
+from ...config import Config
+
 from .tenant import Tenant
 
-asab.Config.add_defaults({
+# "tenant_url" is used to periodically refresh tenants from, expecting "_id" inside a JSON structure,
+# which is compatible with SeaCat Auth product
+Config.add_defaults({
 	'tenants': {
 		'ids': '',
+		'tenant_url': '',  # f. e. http://seacat-auth:8080/tenant
+		'trusted': 0,  # makes sure the tenants are implicitly trusted, even though they are not located in IDs or tenant URL
 	}
 })
 
 
-class TenantService(asab.Service):
+class TenantService(Service):
 
 	def __init__(self, app, service_name="asab.TenantService"):
 		super().__init__(app, service_name)
 		self.App = app
+		self.TenantWebHandler = None
+		self.TenantsTrusted = int(Config['tenants']['trusted'])
 		self.Tenants = {}
-		self.TenantIds = asab.Config['tenants']['ids']
-		self.TenantIds = self.TenantIds.split(',')
 
+		# Load tenants from configuration
+		self.TenantIds = Config['tenants']['ids']
+		self.TenantIds = self.TenantIds.split(',')
 		for tenant_id in self.TenantIds:
-
 			section = 'tenant:params:{}'.format(tenant_id)
-
-			if asab.Config.has_section(section):
-				params = dict(asab.Config.items(section))
+			if Config.has_section(section):
+				params = dict(Config.items(section))
 				self.Tenants[tenant_id] = Tenant(tenant_id, params)
 			else:
 				self.Tenants[tenant_id] = Tenant(tenant_id)
 
+		# Load tenants from URL
+		self.TenantUrl = Config["tenants"]["tenant_url"]
+		if len(self.TenantUrl) > 0:
+			self.LoadTenantsFuture = [asyncio.ensure_future(self._update_tenants(), loop=app.Loop)]
```

Comment:
This shouldn't be a list.

Reply:
Ok, removed it from the variable, but asyncio.wait always expects list...

Reply:
I guess you can do:

`await self.LoadTenantsFuture`